x-pack/solutions/observability/plugins/streams/server/lib/streams/client.ts

```diff
@@ -554,6 +567,17 @@ export class StreamsClient {
     return definition;
   }
 
+  /**
+   * Returns true whether the stream might be ad-hoc created and is not stored in the definition index yet.
+   */
+  private isAdHocUnwiredStream(definition: StreamDefinition): boolean {
+    return (
+      isUnwiredStreamDefinition(definition) &&
+      definition.ingest.routing.length === 0 &&
+      definition.ingest.processing.length === 0
```

Comment:
how does this determine whether the stream is adhoc ? wouldn't a persisted unwired stream also satisfy these conditions ?

Reply:
It does, this is more of a performance optimization. Let me see whether I can refactor this to make it a bit more straightforward